### PR TITLE
chore: Setup Architecture

### LINKED ISSUE #1: Set up development environment
- Configure linter.
- Configure build automation / Maven.
- Configure test runner.
- Set up Continuous Integration
- Etc.

### PR DESCRIPTION
**Pull request by sdsantos**
Fixes #1.

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -22,14 +23,45 @@ android {
...
+
+  // Dependency Injection
+  api 'com.google.dagger:dagger:2.26'
+  kapt 'com.google.dagger:dagger-compiler:2.26'
+
+  // Logging
+  implementation 'com.jakewharton.timber:timber:4.7.1'
+
+  // Testing
   testImplementation 'junit:junit:4.13'
```
Comment (gnarea) [inline app/build.gradle]: Should we drop JUnit4 now that we're integrating JUnit5 in this PR?

Comment (sdsantos) [reply app/build.gradle]: For instrumented tests JUnit5 is experimental and only with some added libraries. But I think JUnit5 is good to go for unit tests.

**Comment by gnarea** [comment]
BTW, now that #20 has been merged, you'll need to edit the title of this PR because there's a GH action checking that PRs use Semantic Release. In this case, you'd need to prefix it with `chore: ` (e.g., `chore: Setup Architecture`).

I think in your case you may only be interested in reading about the list of prefixes/types (by default, it uses these: https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#type).

Apart from generating the version number, we use semantic release so we can automatically generate changelogs like this: https://github.com/relaycorp/relaynet-core-js/releases

**Comment by sdsantos** [comment]
@gnarea right, I had already saved https://www.conventionalcommits.org/en/v1.0.0/ for reading later

**Comment by gnarea** [comment]
@sdsantos, I edited the PR description to add that this PR will fix #1 because I think this is the last change needed to complete that task. Please remove it if that's not correct.

**Comment by sdsantos** [comment]
Now it is ;)

**Pull request merged**